staging/src/k8s.io/apiserver/pkg/registry/rest/update.go

```diff
@@ -136,9 +136,9 @@ func BeforeUpdate(strategy RESTUpdateStrategy, ctx context.Context, obj, old run
 
 	strategy.PrepareForUpdate(ctx, obj, old)
 
-	// ClusterName is ignored and should not be saved
-	if len(objectMeta.GetClusterName()) > 0 {
-		objectMeta.SetClusterName("")
+	// ZZZ_DeprecatedClusterName is ignored and should not be saved
+	if len(objectMeta.GetZZZ_DeprecatedClusterName()) > 0 {
+		objectMeta.SetZZZ_DeprecatedClusterName("")
```

Comment:
issue a warning to the requester here, e.g.:

```go
warning.AddWarning(ctx, "", "metadata.clusterName was specified, but is not preserved and the API field will be removed in 1.25")
```

Reply:
done